### PR TITLE
Implement Interface UserObject and Postprocessor

### PR DESCRIPTION
**Pull request by arovinelli**
- The aim of this new class is to provide an **easy to use** user object to make computation across an interface.

- As mentioned previously this class is not The`InterfaceUserObject` is boundary restricted and allows to make computation on variable across an interface.  Some addition has been done to `FEProblemBase` to allow this new class to properly work and `ComputeUserObjectThread`.
Note taht, becasue we are still waiting on  #12101 this user object is not able to work on material properties at the moment. In fact @lindsayad request #12079 to have a PR without modification tot he material portion of moose. The workaround for this is to convert material property in aux variables and then use the variable as input for `InterfaceUserObject`

- in general the `InterfaceUserObject` is helpful for all that situations in which a UO is needed to do computation across an Interface where the `InterfaceKernel` is used.
continuation of #11660
 ref #11647 

@lindsayad @YaqiWang  @vincentlaboure have helped during development found bags etc, so they already know what this PR is about

### CODE REVIEW

File: large_media [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit c946e65a245948f384835f05a88e6b9a13a1a090
+Subproject commit 6c55e2f9af9616953b98658e909ab9be805ae751
```
[the remark this answers is not in the document]
Comment (arovinelli) [reply large_media]: fixed

File: framework/src/loops/ComputeUserObjectsThread.C [hunk 2]
```diff
@@ -185,6 +189,32 @@ ComputeUserObjectsThread::onInternalSide(const Elem * elem, unsigned int side)
...
+{
+  // Pointer to the neighbor we are currently working on.
+  const Elem * neighbor = elem->neighbor_ptr(side);
+
+  std::vector<UserObject *> userobjs;
+  queryBoundary(Interfaces::InterfaceUserObject, bnd_id, userobjs);
+  if (_interface_user_objects.size() == 0)
+    return;
+
+  _fe_problem.prepareFace(elem, _tid);
```
Comment (arovinelli) [inline framework/src/loops/ComputeUserObjectsThread.C]: @friedmud I saw your reply for #12892 adn there was discussion here #12447 about the condition to apply
what is the right if statement here?

```
 if (!((neighbor->active() && (neighbor->level() == elem->level()) ||
        (neighbor->level() < elem->level())))
```

or something else?

Comment (YaqiWang) [reply framework/src/loops/ComputeUserObjectsThread.C]: We had a discussion but did not conclude this. But I guess a temporary solution is:
```
if (neighbor->active())
  // do your stuff here
else
  mooseError("Interface objects do not support sides with hanging nodes due to mesh adaptivity");
```
@friedmud might want to make a comment here.

**Comment by arovinelli** [comment]
@YaqiWang @permcody @lindsayad there have been a mistake here. I was still finalizing some stuff, actually there was a test failing because of double sum

**Pull request closed without merging**